demo-shell/src/app/components/app-layout/app-layout.component.ts

```diff
@@ -48,6 +50,18 @@ export class AppLayoutComponent {
         { href: '/about', icon: 'info_outline', title: 'APP_LAYOUT.ABOUT' }
     ];
 
-    constructor() {
+    @Input() expandedSidenav = false;
+
+    constructor(private storage: StorageService, private config: AppConfigService) {
+        const preserved = this.storage.getItem('openedSidenav');
```

Comment:
Can you please move this logic into ngOnInit? constructors are supposed to be as "flat" as possible.

And I'm not sure this logic is correct. 
First we would need to check the config, if we need to read from localstorage. If so we can read from localstorage, and set the value if it is already persisted in the local storage, otherwise use the config's openedSidenav value (if it is set!!!) at last but not least, the default value of the component's.